### ISSUE TITLE
Make our event listeners passive, for smoother scrolling

### ISSUE THREAD

**Issue by gnprice** [zulip/zulip-mobile#3440]
Our message list generally scrolls quite smoothly for me, on my Pixel 2 XL. But we sometimes hear reports that it's not.

And, it turns out, there is a simple thing we can do that should improve scrolling! Most of our event handlers should be "passive". See [explainer doc](https://github.com/WICG/EventListenerOptions/blob/gh-pages/explainer.md) on the feature for description and rationale. For specifics, see MDN API doc on [addEventListener](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener).

NB this feature apparently gained support in Safari only with iOS 10 (per the [compatibility table](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#Browser_compatibility) of that last doc.) That is over 99% of our iOS users; and virtually all our Android users are covered, because Chrome auto-updates. So it's OK that the benefit of this will only be for users on those newer browsers -- but we do support iOS 9, so we should make sure we don't just break, by doing feature detection as described in those docs.

**Comment by agrawal-d** [comment]
@zulipbot claim

**Comment by agrawal-d** [comment]
@gnprice zulipbot doesn't seem to have assigned this issue to me. Am I doing something wrong here? This issue is listed as a 'good first issue' by GitHub.

**Comment by agrawal-d** [comment]
**@zulipbot** claim

**Comment by agrawal-d** [comment]
@zulipbot claim

**Comment by agrawal-d** [comment]
@gnprice could you please assign this issue to me?

**Comment by agrawal-d** [comment]
@ray-kraesig it seems @gnprice  is on vacation, could you please assign me to this issue?

**Comment by agrawal-d** [comment]
@ray-kraesig it seems @gnprice  is on vacation, could you please assign me to this issue?

**Comment by ray-kraesig** [comment]
> @ray-kraesig it seems @gnprice is on vacation, could you please assign me to this issue?

Done!

Note that the commentary in the OP concerning what devices we support is out of date. Per [our current documentation](https://github.com/zulip/zulip-mobile/blob/master/docs/architecture/platform-versions.md), we currently support iOS 10 and Android Lollipop (API level 21). (While virtually all of our Android users *do* have updated WebViews, we probably still need to ensure there are no significant regressions on Chrome 36.)

**Comment by agrawal-d** [comment]
@ray-kraesig I have created a pull request ( #3736 ) that should handle this. How do I get someone to review it?

**Comment by gnprice** [comment]
@ray-kraesig writes at https://github.com/zulip/zulip-mobile/pull/3736#issuecomment-572182178 :

> I was just about ready to merge... and then I found this:
> 
> https://developers.google.com/web/updates/2017/01/scrolling-intervention
> https://developer.apple.com/library/archive/releasenotes/General/WhatsNewInSafari/Articles/Safari_11_1.html

Very interesting!

So the story seems to be that on modern browsers, this just happens by default -- there's no need for us to do anything.


> This indicates that, at the time #3440 was filed (2019-04-01), the vast majority of our users should already have been on a default-passive version of Chrome (56.0.2924, released 2017-02-01) or Safari (11.1, released with iOS 11.3, 2018-03-29).
> 
> @gnprice: you mentioned in #3440 that users had been complaining about scrolling being slow. Was that contemporary to the bug being filed, or were those complaints a collection of older issues?

My first reaction was "gosh, I wish I'd linked to the complaints in the issue, oops". :slightly_smiling_face: I went searching through the #mobile history for mentions of scrolling around that time and earlier.

I think in fact the story is that what prompted me to file this issue is that I ran across this "passive event listeners" feature while investigating something else. I remembered "oh, that's a problem we've sometimes heard about" and thought "so here's the prescribed solution -- cool, thanks browser-implementors, I guess we should do that." The complaints I had in mind were far enough back -- a few months at least? -- that I didn't have them handy, and I guess I didn't take the time to go digging for them.

Here's one I might have been thinking of, from 2018-09:
https://chat.zulip.org/#narrow/stream/48-mobile/topic/Feedback/near/644953
Looking at it, I think it's likely that isn't related to this.

That probably wasn't the *only* one I had in mind, but I'm not finding another.

When I filed this, I thought it was describing the best practice as recommended by the browser vendors -- so something we should do, since it's easy, even lacking real firm evidence of its effect. It looks like we've learned it isn't! And that it should have no useful effect. So it looks like it wouldn't make sense for us to do it after all.

Thanks Ray for spotting that, and thanks @agrawal-d for implementing the change! Sorry it turned out not to be a useful change to make -- but going through it was certainly useful for us, in that we learned we don't need to worry that we're neglecting this. And hopefully the review process was useful for you to learn some things, too.
